qiskit/algorithms/amplitude_amplifiers/amplification_algorithm.py

```diff
@@ -0,0 +1,102 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The interface for amplification algorithms and results."""
+
+from abc import ABC, abstractmethod
+from typing import Optional, Any
+
+from .amplification_problem import AmplificationProblem
+from ..algorithm_result import AlgorithmResult
+
+
+class AmplificationAlgorithm(ABC):
```

Comment:
Should this `AmplitudeAmplifier` which would fit more with the naming of the interfaces we have generally used elsewhere given the category (folder) the algorithms are grouped by i.e. `amplitude_amplifiers` in this case. The Result named accordingly too.

Reply:
Oh, of course!